src/components/CandidateModal/index.jsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { CMS_BASE_URL } from 'constants/api';
+
+const CandidateModal = ({ candidate }) => {
+  const getDesiredDetail = key => {
+    let desiredDetail;
+    candidate.details.map(detail => {
+      if (detail.title === key) {
+        desiredDetail = detail;
+        return '';
+      }
+      return '';
+    });
+    return desiredDetail;
+  };
+
+  const displayPriority = priority => {
+    if (!priority) {
+      return '';
+    }
+    return priority.text;
+  };
+
+  const contactMethodList = [
+    'Phone',
+    'Email',
+    'Website',
+    'Twitter',
+    'Facebook',
+    'Instagram',
+    'Youtube',
+    'Other'
+  ];
+
+  const displayContact = contactMethodList.map(cmItem => {
+    let contactMethod = ``;
+    let key = `${cmItem}`;
+    const found = candidate.contacts.find(
+      contact => contact.contactMethod === cmItem
+    );
+    if (found) {
```

Comment:
Can we use a `switch` statement? I think it'll be easier to understand.

Reply:
unable to resolve this at this moment